string/reverse_words.py

```diff
@@ -1,29 +1,9 @@
 
-def reverse(array, i, j):
-    while i < j:
-        array[i], array[j] = array[j], array[i]
-        i += 1
-        j -= 1
-
-def reverse_words(string):
-    arr = list(string)
-    n = len(arr)
-    reverse(arr, 0, n-1)
-
-    start = None
-    for i in range(n):
-        if arr[i] == " ":
-           if start:
-                reverse(arr, start, i-1)
-                start = None
-        elif i == n-1:
-            if start:
-                reverse(arr, start, i)
-        else:
-            if not start:
-                start = i
-    return "".join(arr)
-
+def reverse_words(str):
+	res = " "
```

Comment:
The algorithm uses an extra variable to store the reversed string. A better algorithm would be to reverse the string in place.